### ISSUE TITLE
Installing on Raspbian bookworm failed

### ISSUE THREAD

**Issue by actraiser30** [GeorgeMcMullen/rp-box86wine#4]
Hello

I'm trying to install rp-box86wine on Raspbian bookworm but installation fails on mesa
I attached a photo with the error messages if you can help and update/fix the installer

Thanks 
![IMG_20231109_131155_090_LL](https://github.com/GeorgeMcMullen/rp-box86wine/assets/38106933/e6e2c9c7-d7f4-4da5-9863-792846269513)

**Comment by GeorgeMcMullen** [comment]
Hello,

I've been struggling with the more recent versions of Mesa as well, but mostly on older operating systems. Considering Bookwork seems to be a new OS version, it might be better to not even compile Mesa yourself and just use the OS supplied versions. I'm considering removing the requirement for an entirely separate Mesa install anyway because of the difficulty in compiling for older OSes (especially 32-bit OSes) as well as the existence of precompiled binaries (from https://launchpad.net/~oibaf/+archive/ubuntu/graphics-drivers and https://launchpad.net/~kisak/+archive/ubuntu/kisak-mesa).

What you could possibly do is edit the box86.sh file to remove the Mesa requirement. You could also take a look at the Arm64 branch, where I am experimenting with getting Box86 running on Arm64 OSes for RetroPie. This is a work in progress - the Box86 script has some lines in it which need to be uncommented out.

Let me know how things turn out.

- George

**Comment by actraiser30** [comment]
Hi, don't remove the requirements if possible,just add a switch that detects the os version and install it with apt or build it as in buster it requires to be build,also uncommenting the src in
apt.list file it looks like the installer  takes more required packages
than before,and the building after goes more further than before but still
stops after a while.

**Comment by GeorgeMcMullen** [comment]
@actraiser30 a switch for the OS version might work, especially since I don't yet know what version of Mesa Bookworm is running, I also don't think that the repos I've mentioned support 32-bit OSs (though I will double check this when I have time).

Having a separate version of Mesa prevents Box86/Wine from being added to the official RetroPie installation. That makes sense, because it's yet another thing they would need to support outside of the standard operating system.

When you mention uncommenting the src in apt.list, which src are you talking about?

Also can you run the following command and post the output:

`dpkg -l | grep -i mesa`

Thanks!

**Comment by actraiser30** [comment]
Hello

I mean this file /etc/apt/sources.list
It has an src source inside commented if you uncomment it,it get more files required for mesa building but still the installer stops.

Box86 and wine are installed fine the only issue is mesa,I also managed to play some games, specific, age of mythology and age of empires 3 that they play with issues in buster and bullseye (the main menu is wrong displayed or not 
displayed at all) it's a font issue.

Do you know how to fix that issue in this two games in buster or bullseye?

but in bookworm everything is fine.
Did some tests changing the mesa and wine versions in buster it worked but did not noticed anything different probably my idea.

I will try the command this days and let you know

**Comment by GeorgeMcMullen** [comment]
If it's a font issue, have you installed the required fonts? I've seen games that have use TTF fonts that needed to be installed in Wine. Also there is the possibility that other libraries are required by the games themselves. You might be able to find out more information on that from the Wine website.

**Comment by actraiser30** [comment]
Yes I installed the fonts inside wine, all the package did not work,have
you encounter this issue and fix it??

**Comment by actraiser30** [comment]
Also ask for help in the box86wine GitHub could not help me.

**Comment by actraiser30** [comment]
I run the game in software mode without OpenGL and the font was displayed
correct but the game was very slow and unplayable.

**Comment by GeorgeMcMullen** [comment]
> Yes I installed the fonts inside wine, all the package did not work,have you encounter this issue and fix it??

I've experienced some issues with fonts, but they were resolved by copying the TTF files to `~/.wine/drive_c/windows/Fonts/`, which seems to be what you've already tried.

I checked out the Wine compatibility list for Age of Empires III (https://appdb.winehq.org/objectManager.php?sClass=version&iId=3795) and there are several instructions there, but the one that really resonated with your comment about running it in software mode was _**"Turn off 'allow pixel shader' in winecfg under Graphics"**_. Have you tried that yet? There's another few notes about VSYNC there, but I'm not sure what that means.

**Comment by actraiser30** [comment]
Thanks I try that,were I can get the ttf fonts files?? In case I missed
something

**Comment by GeorgeMcMullen** [comment]
I'm not fully sure, but I saw in an _old_ (really old) Wine bug that it was suggested to use Winetricks to install `corefonts`. Here's the bug, which also references Age of Empires.

https://bugs.winehq.org/show_bug.cgi?id=20456

**Comment by actraiser30** [comment]
There is not such an option "pixel shader" in winecfg under Graphics

there is not really an installer for corefonts I found a Linux deb package with corefonts and extracted the fonts manual in the location you specified,but nothing changed, also installed age of mythology under wine.the same issue as age of empires 3 strange must use the same engine.

**Comment by actraiser30** [comment]
I tried everything they describe in the issues nothing worked, still I
don't have understand how it works on bookworm correct,maybe an extra
package it installs??
